plugins/module_utils/job.py

```diff
@@ -76,9 +76,47 @@ def _get_job_output(job_id="*", owner="*", job_name="*", dd_name=""):
                 str(rc), str(err)
             )
         )
-    if not out:
-        raise RuntimeError("Failed to retrieve job output. No job output found.")
-    jobs = _parse_jobs(out)
+    jobs = []
+    if out:
+        jobs = _parse_jobs(out)
+    if not jobs:
+        jobs = _job_not_found(job_id, owner, job_name, dd_name)
+
+    return jobs
+
+
+def _job_not_found(job_id, owner, job_name, dd_name):
+    jobs = []
+
+    job = {}
+
+    job["job_id"] = job_id
+    job["job_name"] = job_name
+    job["subsystem"] = "NOTFOUND"
+    job["system"] = "NOTFOUND"
+    job["owner"] = owner
+
+    job["ret_code"] = {}
+    job["ret_code"]["msg"] = "Job not found"
```

Comment:
I question what we should have here for a job that does not exist in terms of ret_code, 

By the definition of the property:

- msg - would would be the job error such as: "JCL ERROR" | "ABEND(S|U)xxxx" | "CONV ABEND" | "SEC ERROR" | "SYS FAIL"| CANCELED"| "CC xxxx"

- msg_code - would the string value of the job error reduced down to something more consumable such as "00" ("00" | "04" | "08" | "S0C4"
- msg_txt  - is a place holder for us such that if we can deduce from the msg_code some message that is of value and known to us but does not come from z/OS or ZOAU we can inject it here to help users. For example:
"The target is unreachable"
"Error code xxxx received, please check yyyy is installed and configured"
- code - if and only if we can convert msg_code into an integer value we return it otherwise we return None
Something like msg_code :"00" can be converted to type int: code : 00
Something like msg_code :"ABEND", msg_code :"ABEND(S1234)", msg_code :"CANCELED", can not convert thus return None

JSON mockup for ret_code:
          "ret_code":{    
"msg":" CANCELED",  // "JCL ERROR" | "ABEND(S|U)xxxx" | "CONV ABEND" | "SEC ERROR" | "SYS FAIL"| CANCELED"| "CC xxxx"     
"msg_code":"00",        // "00" | "04" | "08" | "S0C4"      
"msg_txt":"Some message we can deduce from the return code",      
"code":04   
          }
          
 We can discuss, my thoughts are to keep it aligned to the above definitions so maybe:
 msg = 'JOB NOT FOUND'
 msg_txt = 'Unable to find a job matching {job_name} and {job_id}'
 code = None
 msg_code = "None" //unsure of this idea but can't think of a better suggestion.

Reply:
so, in job not found, should owner be "NONE" or None?